### DISCUSSION TITLE
Illegal Instruction error when running llama-server

### DISCUSSION THREAD

**Discussion by AngelaZhang3913** [ggml-org/llama.cpp#8641] [Q&A]
I'm trying to run llama-server using  `./llama-server -m models/codellama-7b.Q4_K_M.gguf -c 2048` after building it. I'm getting an `Illegal Instruction` error message that I can't figure out how to resolve.

I went into gdb and found the illegal instruction, which seems to be this:
`0x000000000045e3bc in void (anonymous namespace)::tinyBLAS<16, float __vector(16), float __vector(16), unsigned short, float, float>::gemm<5, 2>(long, long, long, long) ()`

I've tried doing `export CFLAGS="-march=native -mtune=native"` and `export CXXFLAGS="$CFLAGS"` and rebuilding but it didn't work.

Here is my lscpu output:
Architecture:          x86_64
CPU op-mode(s):        32-bit, 64-bit
Byte Order:            Little Endian
CPU(s):                32
On-line CPU(s) list:   0-31
Thread(s) per core:    2
Core(s) per socket:    8
Socket(s):             2
NUMA node(s):          2
Vendor ID:             GenuineIntel
CPU family:            6
Model:                 79
Model name:            Intel(R) Xeon(R) CPU E5-2620 v4 @ 2.10GHz
Stepping:              1
CPU MHz:               2016.174
CPU max MHz:           3000.0000
CPU min MHz:           1200.0000
BogoMIPS:              4199.74
Virtualization:        VT-x
L1d cache:             32K
L1i cache:             32K
L2 cache:              256K
L3 cache:              20480K
NUMA node0 CPU(s):     0-7,16-23
NUMA node1 CPU(s):     8-15,24-31
Flags:                 fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc aperfmperf eagerfpu pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch epb cat_l3 cdp_l3 invpcid_single intel_ppin intel_pt ssbd ibrs ibpb stibp tpr_shadow vnmi flexpriority ept vpid fsgsbase tsc_adjust bmi1 hle avx2 smep bmi2 erms invpcid rtm cqm rdt_a rdseed adx smap xsaveopt cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local dtherm ida arat pln pts md_clear spec_ctrl intel_stibp flush_l1d

Any help would be appreciated!

**Comment by jgerstle136** [comment]
Got this as well
